### PR TITLE
Fix NodeScheduler not to schedule on any coordinator

### PR DESCRIPTION
**Pull request by zhenxiao**
Fix https://github.com/facebook/presto/issues/2868

### CODE REVIEW

**Comment by zhenxiao** [comment]
@electrum @nileema my understand is there could be more than 1 coordinator in the cluster, and not (or intentionally) to schedule on any of them. Fix me if my understanding is incorrect. Your comments are appreciated

File: presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java [hunk 1]
```diff
@@ -50,7 +50,7 @@
 
 public class NodeScheduler
 {
-    private final String coordinatorNodeId;
+    private final List<String> coordinatorNodeIdList;
```
Comment (yuananf) [inline presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java]: Is it better to use a map?

File: presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java [hunk 2]
```diff
@@ -67,7 +67,9 @@
     public NodeScheduler(NodeManager nodeManager, NodeSchedulerConfig config, NodeTaskMap nodeTaskMap)
     {
         this.nodeManager = nodeManager;
-        this.coordinatorNodeId = nodeManager.getCurrentNode().getNodeIdentifier();
+        this.coordinatorNodeIdList = nodeManager.getCoordinators().stream()
```
Comment (nileema) [inline presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java]: rename to `coordinatorNodeIds`
extract this code snippet to a method and reuse below.

**Comment by nileema** [comment]
Minor comments. 
Looks good otherwise. I'll merge this.

**Comment by zhenxiao** [comment]
@nileema Thank you, I get comments addressed

**Comment by zhenxiao** [comment]
@nileema @electrum kindly ping

**Comment by dain** [comment]
Merged, thanks!

**Pull request closed without merging**